src/parse/validator.rs

```diff
@@ -107,9 +107,11 @@ impl<'help, 'app, 'parser> Validator<'help, 'app, 'parser> {
                 let ok = if arg.is_set(ArgSettings::IgnoreCase) {
                     arg.possible_vals
                         .iter()
-                        .any(|pv| pv.eq_ignore_ascii_case(&val_str))
+                        .any(|ArgValue { value, .. }| value.eq_ignore_ascii_case(&val_str))
                 } else {
-                    arg.possible_vals.contains(&&*val_str)
+                    arg.possible_vals
+                        .iter()
+                        .any(|ArgValue { value, .. }| value == &&*val_str)
```

Comment:
You do this all over the place. Why do we need to destructure? Can't we do something like the following?

```rust
|av| av.value == &&*val_str
```